### DISCUSSION TITLE
how to add indentation for array element

### DISCUSSION THREAD

**Discussion by YK-Unit** [ruby/psych#594] [Q&A]
When psych dump object to yaml file, I hope psych can add indentation for array element, for example:
``` yaml
parent:
  - child
  - child
```
And yet it is：

```yaml
parent:
- child
- child
```

So I wonder how to  add indentation for array element.  Look forward to your solution, thanks very much.

**Comment by NickHackman** [comment]
Seconding this, it would be nice to have support for array indentation following the `indentation` option passed to `dump` or `to_yaml` would be ideal. If this already exists I haven't been able to find any sort of documentation referencing it. 

If it needs to be implemented then I'd be interested in implementing this feature :+1:

**Comment by Paxa** [comment]
It looks much better with indentation

**Comment by cvirtucio** [comment]
any update on this?

also the [documentation](﻿https://github.com/ruby/psych/blob/v5.1.2/lib/psych.rb#L110-L115) seems incorrect. the `dump` class method creates a [YAMLTree visitor](https://github.com/ruby/psych/blob/v5.1.2/lib/psych.rb#L511) with the options, but I'm not seeing any reference to the `indentation` symbol in any of the usages of the `@options` field: https://github.com/ruby/psych/blob/v5.1.2/lib/psych/visitors/yaml_tree.rb#L105-L118

**Comment by searls** [comment]
Unfortunately, hugo's YAML frontmatter parser will barf on non-indented array items which is a little bit of a blocker to my running scripts against my blog content with ruby.
